packages/legend-studio/src/components/editor/edit-panel/service-editor/ServiceRegistrationEditor.tsx

```diff
@@ -0,0 +1,232 @@
+        </div>
+      </div>
+      <PanelLoadingIndicator
+        isLoading={registrationState.registrationState.isInProgress}
+      />
+      <div className="panel__content__form">
+        {registrationState.registrationPhase && (
+          <div className="service-registration-editor__progress-msg">
+            {`${prettyCONSTName(registrationState.registrationPhase)}...`}
+          </div>
+        )}
+        <div className="panel__content__form__section">
+          <div className="panel__content__form__section__header__label">
+            Activate Service
+          </div>
+          <div
+            className="panel__content__form__section__toggler"
+            onClick={toggleActivatePostRegistration}
+          >
+            <button
+              type="button"
+              className={clsx('panel__content__form__section__toggler__btn', {
+                'panel__content__form__section__toggler__btn--toggled':
+                  registrationState.activatePostRegistration,
+              })}
+              tabIndex={-1}
+            >
+              {registrationState.activatePostRegistration ? (
+                <CheckSquareIcon />
+              ) : (
+                <SquareIcon />
+              )}
+            </button>
+            <div className="panel__content__form__section__toggler__prompt">
+              Activates service after registering.
+            </div>
+          </div>
+        </div>
+        <div className="panel__content__form__section">
+          <div className="panel__content__form__section__header__label">
+            Alloy Server
+          </div>
+          <div className="panel__content__form__section__header__prompt">
+            The server where your service will be registered
+          </div>
+          <CustomSelectorInput
+            options={envOptions}
+            onChange={onServerEnvChange}
+            value={selectedEnvOption}
+            darkMode={true}
+          />
+        </div>
+        <div className="panel__content__form__section">
+          <div className="panel__content__form__section__header__label">
+            Service Type
+          </div>
+          <div className="panel__content__form__section__header__prompt">
+            The kind of service you want to register. Used to determine where
+            the metadata will be fetched from.
+          </div>
+          <CustomSelectorInput
+            options={serviceTypesOptions}
+            onChange={onServiceTypeSelectionChange}
+            value={selectedServiceType}
+            darkMode={true}
+          />
+        </div>
+        <div className="panel__content__form__section">
+          <div className="panel__content__form__section__header__label">
+            Project Version
+          </div>
+          <div className="panel__content__form__section__header__prompt">
+            The version of your project you want to use for registration. Only
+            relevant for semi and prod service types.
```

Comment:
Let's remove the period

Reply:
Also, let's say `...semi-interactive and production services.` instead of `semi and prod service types`.